Export Analysis Reports/Sdl.Community.ExportAnalysisReports/Service/ReportService.cs

```diff
@@ -0,0 +1,355 @@
+		}
+
+		public string JsonPath => Path.Combine(_communityFolderPath, "ExportAnalysisReportSettings.json");
+		public string ReportsFolderPath { get; set; }
+		/// <summary>
+		/// Check if the report was successfully generated
+		/// </summary>
+		/// <param name="reportOutputPath"></param>
+		/// <param name="isChecked"></param>
+		/// <param name="optionalInformation"></param>
+		/// <param name="projects"></param>
+		/// <returns></returns>
+		public bool GenerateReportFile(BindingList<ProjectDetails> projects, OptionalInformation optionalInformation, string reportOutputPath, bool isChecked)
+		{
+			try
+			{
+				_help.CreateDirectory(reportOutputPath);
+				var projectsToBeExported = projects.Where(p => p.ShouldBeExported).ToList();
+				var areCheckedLanguages = projectsToBeExported.Any(p => p.PojectLanguages.Any(l => l.Value));
+				if (!areCheckedLanguages && projectsToBeExported.Count >= 1)
+				{
+					_messageBoxService.ShowInformationMessage(PluginResources.SelectLanguage_Export_Message, PluginResources.ExportResult_Label);
+					return false;
+				}
+				foreach (var project in projectsToBeExported)
+				{
+					// check which languages to export
+					if (project.PojectLanguages != null)
+					{
+						var checkedLanguages = project.PojectLanguages.Where(l => l.Value).ToList();
+
+						foreach (var languageReport in checkedLanguages)
+						{
+							if (string.IsNullOrEmpty(project.ReportPath))
+							{
+								project.ReportPath = reportOutputPath;
+							}
+
+							WriteReportFile(project, optionalInformation, languageReport, isChecked);
+						}
+					}
+				}
+				return true;
+			}
+			catch (Exception exception)
+			{
+				Log.Logger.Error($"GenerateReport method: {exception.Message}\n {exception.StackTrace}");
+				throw;
+			}
+		}
+
+		/// <summary>
+		/// Get report output path from Json file
+		/// </summary>
+		/// <param name="jsonPath"></param>
+		/// <returns></returns>
+		public string GetJsonReportPath(string jsonPath)
+		{
+			if (File.Exists(jsonPath))
+			{
+				using (var r = new StreamReader(jsonPath))
+				{
+					var json = r.ReadToEnd();
+					var item = JsonConvert.DeserializeObject<JsonSettings>(json);
+					if (!string.IsNullOrEmpty(item.ExportPath))
+					{
+						return item.ExportPath;
+					}
+				}
+			}
+			return string.Empty;
+		}
+
+		/// <summary>
+		/// Check if the exported path is the same
+		/// </summary>
+		/// <param name="reportOutputPath"></param>
+		/// <returns></returns>
+		public bool IsSameReportPath(string reportOutputPath)
+		{
+			var jsonReportPath = GetJsonReportPath(JsonPath);
+			if (!string.IsNullOrEmpty(jsonReportPath) && !string.IsNullOrEmpty(reportOutputPath) && jsonReportPath.Equals(reportOutputPath))
+			{
+				return true;
+			}
+
+			return false;
+		}
+
+		/// <summary>
+		/// Check if the report file exists
+		/// </summary>
+		/// <param name="reportFolderPath"></param>
+		/// <returns></returns>
+		public bool ReportFileExist(string reportFolderPath)
```

Comment:
>if inversions< recommended

Reply:
More if inversions were applied on the report service and studio service